config/runtime/server.go

```diff
@@ -627,20 +631,48 @@ func getPortsHostsList(hosts []string, defaultPort int) (Ports, error) {
 	return portsHosts, nil
 }
 
-func newEndpointMap(srvConf *config.Server) endpointMap {
+func newEndpointMap(srvConf *config.Server, serverOptions *server.Options) (endpointMap, error) {
 	endpoints := make(endpointMap)
 
+	apiBasePaths := make(map[string]struct{})
+
 	for _, api := range srvConf.APIs {
+		basePath := serverOptions.APIBasePaths[api]
+
+		var filesBasePath, spaBasePath string
+		if serverOptions.FilesBasePath != "" {
+			filesBasePath = path.Join("/", serverOptions.FilesBasePath)
```

Comment:
`NewServerOptions` ensures a leading `/` for all paths (via `options.SrvBasePath`).
This extra joins seems to be obsolete here.

Reply:
Done